File: spring-core-lab-solutions/Lab-07-solution/petclinic/src/main/java/com/petclinic/core/VisitService.java
package com.petclinic.core;

import jakarta.persistence.EntityNotFoundException;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class VisitService {
    private final VisitRepository visitRepository;

    public VisitService(VisitRepository visitRepository) {
        this.visitRepository = visitRepository;
    }

    public Visit findById(Integer id) {
        Optional<Visit> visit = this.visitRepository.findById(id);
        if(visit.isEmpty()) {
            throw new EntityNotFoundException("entity " +id + " was not found");
        }
        else return visit.get();
    }

    public Visit findByReferenceNumber(String referenceNumber) {
        return visitRepository.findByReferenceNumber(referenceNumber);
    }

    public void delete(Visit visit) {
        this.visitRepository.delete(visit);
    }

    public void save(Visit visit) {
        this.visitRepository.save(visit);
    }
}

File: spring-core-lab-solutions/Lab-kotlin/src/main/kotlin/com/petclinic/core/VisitService.kt
package com.petclinic.core

import org.springframework.stereotype.Service

@Service
class VisitService(
        val visitRepository:VisitRepository) {
    fun save(visit: Visit): Visit {
        return visitRepository.save(visit)
    }

    fun findById(id: Int): Visit {
        return visitRepository.findById(id).orElseThrow()
    }
}